### PR TITLE
Require py37

### PR DESCRIPTION
**Pull request by escapewindow**
This adds support for taskcluster-client.py 4.0.0 (taskcluster.aio rather than taskcluster.async), and makes py37 a required test.

### CODE REVIEW

**Comment by escapewindow** [comment]
The py37 bustage should theoretically be fixed by #244... trying a rerun.

File: .travis.yml [hunk 1]
```diff
@@ -4,11 +4,13 @@ matrix:
   fast_finish: true
   include:
     - python: 3.6
+      dist: trusty
       env: TOXENV=py36 NO_TESTS_OVER_WIRE=1
-    - python: 3.7-dev
+      sudo: false
+    - python: 3.7
+      dist: xenial
```
Comment (Callek) [inline .travis.yml]: probably worth a comment on "why xenial" ;-) (I know this answer)

Comment (escapewindow) [reply .travis.yml]: Done. Also now require taskcluster>=4 in setup.py and requirements-dev, and added py37 to setup.py.

**Comment by escapewindow** [comment]
Reran another intermittent that would theoretically be fixed by #244

**Comment by escapewindow** [comment]
Hmmmm, close and reopen?

**Pull request merged**